test/runnable/test11934.d

```diff
@@ -0,0 +1,25 @@
+void main()
+{
+    import core.memory : GC;
+
+    struct Struct11934
+    {
+        this(int i) { instances++; }
+        this(this) { instances++; }
+        ~this() { instances--; }
+        static size_t instances = 0;	
+    }
+
+    struct Range11934
+    {
+        void popFront() { cnt++; }
+        @property front() { return Struct11934(0); }
+        @property empty() { return cnt >= 10; }
+        size_t cnt;
+    }
+
+    foreach(ref i; Range11934()) { }
+
+    GC.collect();
```

Comment:
There should be no need for calling the `GC.collect` here. `struct` are allocated on the stack and destroyed deterministically once you leave a scope (i.e. after the foreach).